docs/user-guide.md

```diff
@@ -752,6 +753,21 @@ The Spark Operator provides limited support for resource quota enforcement using
 
 If you are running Spark applications in namespaces that are subject to resource quota constraints, consider enabling this feature to avoid driver resource starvation. Quota enforcement can be enabled with the command line arguments `-enable-resource-quota-enforcement=true`. It is recommended to also set `-webhook-fail-on-error=true`.
 
+## Running multiple instances of the operator within the same k8s cluster
+
+If you need to run multiple instances of the operator within the same k8s cluster. Therefore, you need to make sure that the running instances should not compete for the same custom resources or pods. You can achieve this:
+
+Either:
+* By specifying a different `namespace` flag for each instance of the operator.
+
+OR if you want your operator to watch specific resources that may exist on different namespaces:
+
+* You need to add custom labels on resources by defining for each instance of the operator a different set of labels in `-label-selector-filter (e.g. env=dev,app-type=spark)`.
+* Run different `webhook` instances by specifying different `-webhook-config-name` flag for each deployment of the operator.
+* Specify different `webhook-svc-name` and/or `webhook-svc-namespace` for each instance of the operator. 
+* Edit the job that generates the certificates `webhook-init` by specifying the namespace and the service name of each instance of the operator, `e.g. "/usr/bin/gencerts.sh", "-n", "ns-op1", "-s", "spark-webhook-op1", "-p"`. Where `spark-webhook-op1` should match what you have specified in `webhook-svc-name`.
+* (Optional) Although resources are already filtred with respect to the specified labels on resources. You may also specify different labels in `-webhook-namespace-selector` and attach these labels to the namepsaces on which you want the webhook to listen to.
```

Comment:
This reminds me the following question: given that `webhook-namespace-selector` allows the webhook to filter namespaces by labels on them, can we do the same for custom resources and pods and filter them by labels on namespaces also, instead of having a different semantic of filtering? This way, it's less confusing to users.

Reply:
I investigated this option. But it seems to be not possible or at least not supported natively. The type webhook proposes a [namespaceSelector](https://github.com/kubernetes/api/blob/master/admissionregistration/v1beta1/types.go#L234) which allows filtering natively on namespaces using labels. All we have to do is to specify the list of labels in `-webhook-namespace-selector`. This changes with regard to the custom resources/pods and I don't see a straight solution to do filtering based on lables attached to namsepaces. I tried to set labels on pods/crs the same as the ones set on the namespace. But pods/crs does not inherit theses labels and no filtering happens.

The cleanest way IMHO, Filtering should be done using `informers.WithNamespace` and/or `informers.WithTweakListOptions`. The latter exposes a list of possible options to filter on, which we can find [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L335). Labels gives a clean and finer way to filter resources even within the namespace itself. For this reason and as per my understanding, we really need to seperate filtering semantic: NamespaceSelector on namespace for the webhook and labelSelector on resources for crs/pods.